src/util/Spotify.js

```diff
@@ -0,0 +1,90 @@
+let userToken; 
+let expTime;
+const clientID = '';
+const redirectURI = '';
+
+
+module.exports = {
+
+    getAccessToken: function() {
+        const pageLoc = window.location.href;
+
+        if (userToken) {
+            console.log(userToken)
+            return userToken;
+        };
+        
+        const userTokenMatch = pageLoc.match(/access_token=([^&]*)/);
+        const expTimeMatch = pageLoc.match(/expires_in=([^&]*)/);
+        
+        if (userTokenMatch && expTimeMatch) {
+            userToken = userTokenMatch[1];
+            expTime = Number(expTimeMatch[1]);
+            window.setTimeout(() => userToken = '', expTime * 1000);
+            window.history.pushState('Access Token', null, '/');
+            return userToken;
+        } else {
+            const accessUrl = 'https://accounts.spotify.com/authorize?client_id=' + clientID + '&response_type=token&scope=playlist-modify-public&redirect_uri=' + redirectURI;
+            window.location = accessUrl;
+        }
+    },
+
+    spotifySearch(term) {
+        const accessToken = module.exports.getAccessToken();
+        return fetch('https://api.spotify.com/v1/search?type=track&q=' + term, 
+            { headers: {
+                Authorization: 'Bearer ' + accessToken
+            }
+        }).then(response => {
+            return response.json()
```

Comment:
Since you are doing multiple `fetch().then(response => response.json())`, you can write a `fetchJson()` function that does both of these calls so you don't have to repeat them everywhere.

When you have an arrow function with only one line, don't use {}, just return directly, like example on this comment.